src/main/java/org/folio/helper/PurchaseOrderLineHelper.java

```diff
@@ -672,47 +626,43 @@ private Future<String> generateLineNumber(CompositePurchaseOrder compOrder, Requ
 
   private Future<CompositePoLine> populateCompositeLine(PoLine poline, RequestContext requestContext) {
     return purchaseOrderLineService.operateOnPoLine(HttpMethod.GET, poline, requestContext)
-      .compose(compPol -> getLineWithInstanceId(compPol, requestContext));  }
+      .compose(compPol -> getLineWithInstanceId(compPol, requestContext));
+  }
 
   private Future<CompositePoLine> getLineWithInstanceId(CompositePoLine line, RequestContext requestContext) {
-    if (!Boolean.TRUE.equals(line.getIsPackage())) {
-      return titlesService.getTitles(1, 0, QUERY_BY_PO_LINE_ID + line.getId(), requestContext)
-        .map(titleCollection -> {
-          List<Title> titles = titleCollection.getTitles();
-          if (!titles.isEmpty()) {
-            line.setInstanceId(titles.get(0).getInstanceId());
-          }
-          return line;
-        });
-    } else {
+    if (Boolean.TRUE.equals(line.getIsPackage())) {
       return Future.succeededFuture(line);
     }
+    var query = String.format(QUERY_BY_PO_LINE_ID, line.getId());
+    return titlesService.getTitles(1, 0, query, requestContext)
+      .map(titleCollection -> {
+        var titles = titleCollection.getTitles();
+        if (!titles.isEmpty()) {
+          line.setInstanceId(titles.get(0).getInstanceId());
+        }
+        return line;
+      });
   }
 
-  private CompositePurchaseOrder addLineToCompOrder(CompositePurchaseOrder compOrder, JsonObject lineFromStorage) {
-    PoLine poLine = lineFromStorage.mapTo(PoLine.class);
-    poLine.setAlerts(null);
-    poLine.setReportingCodes(null);
-    CompositePoLine compPoLine = JsonObject.mapFrom(poLine).mapTo(CompositePoLine.class);
+  private CompositePurchaseOrder addLineToCompOrder(CompositePurchaseOrder compOrder, PoLine lineFromStorage) {
+    var compPoLine = convertToCompositePoLine(lineFromStorage);
```

Comment:
do you think it is not needed?
poLine.setAlerts(null);
poLine.setReportingCodes(null);
Maybe it was specially set to null?

Reply:
I see, they are deleted in convertToCompositePoLine()